src/diffusers/loaders/lora_pipeline.py

```diff
@@ -2113,8 +2258,116 @@ def unfuse_lora(self, components: List[str] = ["transformer", "text_encoder"], *
+                bias = module_bias is not None
+
+                lora_A_weight_name = f"{name}.lora_A.weight"
+                lora_B_weight_name = f"{name}.lora_B.weight"
+
+                if lora_A_weight_name not in state_dict.keys():
+                    continue
+
+                in_features = state_dict[lora_A_weight_name].shape[1]
+                out_features = state_dict[lora_B_weight_name].shape[0]
+
+                if tuple(module_weight.shape) == (out_features, in_features):
+                    continue
+
+                module_out_features, module_in_features = module_weight.shape
+                if out_features < module_out_features or in_features < module_in_features:
+                    raise NotImplementedError(
+                        f"Only LoRAs with input/output features higher than the current module's input/output features "
+                        f"are currently supported. The provided LoRA contains {in_features=} and {out_features=}, which "
+                        f"are lower than {module_in_features=} and {module_out_features=}. If you require support for "
+                        f"this please open an issue at https://github.com/huggingface/diffusers/issues."
+                    )
+
+                logger.debug(
+                    f'Expanding the nn.Linear input/output features for module="{name}" because the provided LoRA '
+                    f"checkpoint contains higher number of features than expected. The number of input_features will be "
+                    f"expanded from {module_in_features} to {in_features}, and the number of output features will be "
+                    f"expanded from {module_out_features} to {out_features}."
+                )
+
+                has_param_with_shape_update = True
+                parent_module_name, _, current_module_name = name.rpartition(".")
+                parent_module = transformer.get_submodule(parent_module_name)
+
+                expanded_module = torch.nn.Linear(
+                    in_features, out_features, bias=bias, device=module_weight.device, dtype=module_weight.dtype
+                )
+
+                new_weight = torch.zeros_like(
+                    expanded_module.weight.data, device=module_weight.device, dtype=module_weight.dtype
+                )
+                slices = tuple(slice(0, dim) for dim in module_weight.shape)
+                new_weight[slices] = module_weight
+                expanded_module.weight.data.copy_(new_weight)
+
+                if bias:
+                    new_bias = torch.zeros_like(
+                        expanded_module.bias.data, device=module_bias.device, dtype=module_bias.dtype
+                    )
+                    slices = tuple(slice(0, dim) for dim in module_bias.shape)
+                    new_bias[slices] = module_bias
+                    expanded_module.bias.data.copy_(new_bias)
+
+                setattr(parent_module, current_module_name, expanded_module)
```

Comment:
Yes, I think okay to remove. Since the bias tensor has the shape of output features, it will not expand since we only want input features to be expanded (which only affects the weights). Thanks for the catch!

Reply:
I pushed some changes to keep it but with better guards. Works for you?

Reply:
Thanks, but since we don't have any case for at the moment, I think we should remove it, no? I initially thought we did need it, but after the latest rounds of testing, we know that the bias does not need expansion. And we don't have any real checkpoints to test this behaviour either, so will let you make the final call.

Reply:
Okay for me!